test/blackbox-tests/test-cases/test-build-if/feature.t

```diff
@@ -1,4 +1,4 @@
-enabled_if has a limitation: it attempts building even if enabled_if evaluates to false.
+enabled_if and build_if have similar behavior
```

Comment:
cc @emillon. This PR seems to fix the case found in #7899.

Reply:
The old behavior was in fact correct and this PR breaks it. The description of this test might be the source of confusion. There's no limitation anywhere, it's just that the different semantics were chosen for `enabled_if` on tests.

Reply:
Just as a side note, if I'm understanding the current behavior properly, it is quite confusing from a user perspective. `enabled_if` semantics are now different depending on where it's used:

- In `executable` stanzas, the field means the executable is not built if it evaluates to `false`
- In `test` stanzas:
  - when `all` alias is used, the field means the test executable is always built, regardless its evaluated value
  - but for `runtest` alias it means the executable will not be built if it evaluates to `false` (like `executable` stanzas)

I found this comment that explains the use case for it: https://github.com/ocaml/dune/pull/7899#issuecomment-1581552347.

It's probably too late, but wonder if it would have been possible to leave the behavior of `enabled_if` consistent, and allow advanced use cases to use a combination of `rule` and `executable`, or add a new field `test_if`.

Reply:
Yes, in hindsight it would be better to do it your way. We could revisit it for 4.0